docs/source/users/problem_definition_files/spinw.rst

```diff
@@ -15,105 +32,132 @@ software, name, description
+  to load the data in appropriately in the `wye_function`. 
+
+function
+  The function is defined by one or two matlab files which returns a SpinW model of the foreground or the foreground and background
+  repectively separated by a semicolon.
+
+  The format is again comma seperated key-value pairs, with the matlab files 
+  defined by the variable "foreground" and "background" and the remaining pairs defining starting
+  values as in the native parser.
+
+  Examples:
+
+    Only foreground: 
+    
+    .. code-block:: rst
+
+      function = 'foreground=m_scripts/functions/mftest_gauss_bkgd.m ,height=100,centre=50,sigma=7,const=0,grad=0'
 
-instrument
-  The instrument defines where the measurements were taken and again requires
-  a ``class`` followed by the correct arguments for the class.
+    Foreground and background:
+
+    .. code-block:: rst
+
+      function = 'foreground=m_scripts/functions/gauss.m ,height=1100,centre=66 ,stdev=13; background=m_scripts/functions/linear_bg.m ,bkgd_const=15'
 
-  As with sample, details can be found `here <https://pace-neutrons.github.io/Horace/v3.6.2/user_guide/Resolution_convolution.html#the-tobyfit-class>`__.
   
-  e.g.::
-    instrument = 'class=maps_instrument,ei=70,hz=250,chopper=S'
+wye_function
+  The wye_function is defined by a matlab file which returns the `w` (this could be a sqw ,d1d, d2d, d3d and d4d object), 
+  `e` (this is the standard deviation arrays from the objects), `y` (intensity arrays from the objects), along with a  
+  mask array (`msk`) that indicates which elements are to be retained (where elements of msk are true, the corresponding elements of 
+  y and e are retained). This matlab file takes in the path of the datafile and the path of where the matlab functions are located.
 
-function
-  The function is defined by a matlab file which returns a SpinW model.
+  Explained example of the wye_function:
 
-  The format is again comma seperated key-value pairs, with the matlab file
-  defined by the variable "matlab_script" and the remaining pairs defining starting
-  values as in the native parser.
+  The first three lines adds the path to matlab functions need for fitting and loads the `w` object.
 
-  e.g.::
-    function = 'matlab_script=pcsmo_model.m,J=35,D=0,gam=30,temp=10,amp=300'
+  .. code-block:: rst
 
-random_fraction_pixels
-  SpinW problems can be very large. This option will mask out a proportion of
-  pixels to reduce the problem size.
+    addpath(genpath(path));
+    source_data = load(datafile);
+    w = source_data.w1 ;
 
-  The following example retains only 1% of the data.
+  The next line gets the y and e from the `w` object. These are the true `y` and `e` values from the experiment.    
 
-  e.g.::
-    random_fraction_pixels = 0.01
+  .. code-block:: rst
+    
+    [spinw_y, spinw_e] = sigvar_get(w);
+
+  The next block of code runs :code:`simulate()` once for the sole purpose of getting the msk array 
```

Comment:
Is this still true? I thought you changed this

Reply:
I've changed it now.